project-koku/koku
# Add order by to bigquery queries (for manifest creation datetimes)

## Jira Ticket

[COST-4975](https://issues.redhat.com/browse/COST-4975)

## Description

This change will add an order by to the GCP bigquery we make so when creating manifests for a new provider they are created in sequential order. There is a future code side affect in cluster info because these are created some what randomly in order 

## Testing

1. Checkout main
2. Restart Koku
3. Add REAL GCP source (Running this QE test is best test_api_ocp_on_gcp_source_cluster_info_infra)
5. If multiple months are ingested this test will likely fail do to the manifest creation order (since latest is previous month when we do lookups for cluster info)
6. Check sources endpoint for status summary (see examples below)
7. Checkout Branch
8. Rerun the above test and it should now pass.
9. check sources endpoint for completed summary state


Example main:
"summary": {
            "state": "Pending"
        }

Branch Example:
"summary": {
            "end": "2024-04-30T10:42:45.011783+00:00",
            "start": "2024-04-30T10:42:43.755820+00:00",
            "state": "complete"
        }

## Release Notes

```markdown
* [COST-4975](https://issues.redhat.com/browse/COST-4975) Add ordering to bigquery to create manifests in asc order
```
